pkg/csi/service/wcp/controller.go

```diff
@@ -460,19 +460,17 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 	}
 	// Fetch the accessibility requirements from the request.
 	topologyRequirement = req.GetAccessibilityRequirements()
+
+	log.Debugf("topologyRequirement: %s", topologyRequirement) // TODO: remove
+
 	filterSuspendedDatastores := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CnsMgrSuspendCreateVolume)
 	isTKGSHAEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TKGsHA)
 	topoSegToDatastoresMap := make(map[string][]*cnsvsphere.DatastoreInfo)
 	if isTKGSHAEnabled {
 		// TKGS-HA feature is enabled
 		// Identify the topology keys in Accessibility requirements.
 		hostnameLabelPresent, zoneLabelPresent = checkTopologyKeysFromAccessibilityReqs(topologyRequirement)
-		// TODO: TKGS-HA: This case will only arise when spherelet will add zone and hostname labels to CSINodes.
-		// Currently spherelet only accepts hostname. We will handle this case later.
-		if zoneLabelPresent && hostnameLabelPresent {
-			return nil, csifault.CSIUnimplementedFault, logger.LogNewErrorCodef(log, codes.Unimplemented,
-				"support for topology requirement with both zone and hostname labels is not yet implemented.")
-		} else if zoneLabelPresent {
+		if zoneLabelPresent {
```

Comment:
for vddp on stretched supervisor we will have both zone and host topology keys in the requirement

Reply:
@divyenpatel yes. In that scenario, we will first get all the datastores in the zone then take an intersection of the accessible nodes if `storage pool` is present in the creation request.

Reply:
This code will also get executed when both host & zone keys are present right?  The `sharedDatastores` you will get at the end of the block are zonal datastores. I checked the code flow, looks like this value does not get used at all in vdpp case. Is that right?

Reply:
I have discussed the flow with Saloni. For vDPp workloads, we can directly use the storage pool to get the datastore URL and provide volume provisioning. I've reverted back these changes.